src/Upgrade/Application/Strategy/ReleaseApp/Processor/ModuleFetcher.php

```diff
@@ -67,30 +77,52 @@ public function require(ModuleDtoCollection $moduleCollection): PackageManagerRe
             return new PackageManagerResponseDto(true, static::MESSAGE_NO_PACKAGES_FOUND);
         }
 
-        return $this->requirePackageCollection($packageCollection);
+        return $this->requirePackageCollection(
+            $this->getPackageManagerPackages($packageCollection),
+        );
     }
 
     /**
      * @param \Upgrade\Domain\Entity\Collection\PackageCollection $packageCollection
      *
+     * @throws \InvalidArgumentException
+     *
+     * @return \Upgrade\Application\Dto\PackageManagerPackagesDto
+     */
+    protected function getPackageManagerPackages(PackageCollection $packageCollection): PackageManagerPackagesDto
+    {
+        foreach ($this->packageManagerPackagesFetchers as $packageManagerPackagesFetcher) {
+            if (!$packageManagerPackagesFetcher->isApplicable()) {
```

Comment:
why not?
`if ($packageManagerPackagesFetcher->isApplicable()) {
    return $packageManagerPackagesFetcher->fetchPackages($packageCollection);
}`

Reply:
For this case is no difference but in more complex should as described above